site/app/templates/admin/admin_gradeable/AdminGradeablePeers.twig

```diff
@@ -90,48 +107,59 @@
+        add_students_div.append('<input class="add_user_ids" type="text" name="add_user_id_0" /><br>');
         $('[name="add_user_id_0"]', form).autocomplete({
             source: student_autofill
         });
         $('[name="add_user_id_0"]').autocomplete( "option", "appendTo", form );
-        add_students_div.append('<p id="add_student_button" style="cursor: pointer;" onclick="addNewStudent();" aria-label="Add More Users"><i class="fas fa-plus-square"></i>Add More Users<br></p>');        
+        add_students_div.append('<p id="add_student_button" style="cursor: pointer;" onclick="addStudentTextbox(\'#admin-gradeable-peers-form\',\'#students-to-add\');" aria-label="Add More Users"><i class="fas fa-plus-square"></i>Add More Users<br></p>');  
     }
 
-    function checkSubmitValid() {
-        let student_autofill = JSON.parse($('#student_autofill').val());
+    function checkNewStudentsValid(grader_id, user_input, valid_students, new_grader) {
+        // check if the newly inputed student is a valid student in the database
+        let student_autofill = JSON.parse($('#student-autofill').val());
         let valid_user_ids = student_autofill.map(tmp => tmp['value']);
-        let found = valid_user_ids.includes($(this).val()) ? true : false;
-        let valid_entry = found || ($(this).val() == '');
-        if (valid_entry) {
-            $('#admin-gradeable-peers-submit').prop('disabled', false);
+        let valid_entry = valid_user_ids.includes(user_input.val());
+        // check if the newly inputed student is already a student for grader
+        // this is automatically false when adding a new grader
+        let student_pairs = JSON.parse($('#peer-grader-pairs').val());
+        let already_exists = !new_grader && student_pairs[grader_id].includes(user_input.val());
+        // the newly inputed student cannot be the grader themself
+        let self_grading = grader_id === user_input.val();
+        // if all 3 above conditions are met, allow user to press submit
+        let index = Number(user_input.attr('name').substr(12));
+        if (user_input.val() == '') {
+            valid_students[index] = undefined;
+        } else if (valid_entry && !already_exists && !self_grading) {
+            valid_students[index] = true;
         } else {
-            $('#admin-gradeable-peers-submit').prop('disabled', true);
+            valid_students[index] = false;
         }
     }
 
     function submitFormAjax() {
-        let url = $('#peers_submit_url').val();
+        let url = $('#peers-submit-url').val();
         let form = $("#admin-gradeable-peers-form");
         let grader_id = $("#admin-gradeable-peers-title").text().substr(18);
         let fd = new FormData();
-        fd.append('csrf_token', $("#csrf_token").val());
+        fd.append('csrf_token', $("#csrf-token").val());
         fd.append('grader_id', grader_id);
         fd.append('remove_grader', $('#removed-all').text());
```

Comment:
Currently you can delete the grader (with all of their students) from the `Edit-Students` form... what about moving this functionality from this form to an action button (trash-icon/ delete button) next to the `Edit-students` row?

Reply:
That's a good idea, but maybe that can be a future PR because it's a more nitpicky than functionality thing (not to be rude haha, thank you for mentioning this though)